lib/components/viewers/pattern-viewer.tsx

```diff
@@ -109,40 +108,25 @@ const PatternViewer = ({
         />
         {/* Header Block */}
         <div
-          className="route-viewer-header"
-          style={{ backgroundColor: routeColor }}
+          className="header-with-back-button pattern-viewer-header"
+          style={{ backgroundColor: backgroundColor }}
         >
-          {/* Back button */}
-          {!hideBackButton && (
-            <div className="back-button-container">
-              <Button bsSize="small" onClick={_backClicked}>
-                <StyledIconWrapper>
-                  <ArrowLeft />
-                </StyledIconWrapper>
-                <FormattedMessage id="common.forms.back" />
-              </Button>
-            </div>
-          )}
-          <div className="header-text route-expanded">
-            <h1 style={{ display: 'contents' }}>
-              {!route.pending && ModeIcon && (
-                <ModeIcon
-                  aria-label={getFormattedMode(
-                    getModeFromRoute(route).toLowerCase(),
-                    intl
-                  )}
-                  mode={getModeFromRoute(route)}
-                  style={{ maxHeight: 40 }}
-                  width={22}
-                />
-              )}
-              <RouteName
-                isOnColoredBackground
+          <BackButton
+            backButtonText={backButtonText}
+            id="pattern-viewer-back-button"
+            onClick={_backClicked}
+          />
+          <h1 style={{ display: 'contents', lineHeight: '1.4' }}>
+            {!route.pending && ModeIcon && (
+              <RouteRowDetails
+                intl={intl}
+                isActive={false}
+                ModeIcon={ModeIcon}
                 route={route}
                 RouteRenderer={RouteRenderer}
               />
-            </h1>
-          </div>
+            )}
+          </h1>
```

Comment:
Is there any way to hide the shadow if there is no overflow? It's quite awkward<img width="543" height="327" alt="Screenshot 2025-09-19 at 5 43 01 PM" src="https://github.com/user-attachments/assets/d4cf7cdf-bb9f-4d89-afc1-9579db3830a6" />

Reply:
You may need to pull the branch? This shadow is gone, thats what this was referring to https://github.com/opentripplanner/otp-react-redux/pull/1467#discussion_r2334527984